lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -25,23 +30,21 @@ function wp_theme_has_theme_json() {
 		 * The reason not to store it as a boolean is to avoid working
 		 * with the $found parameter which apparently had some issues in some implementations
 		 * https://developer.wordpress.org/reference/functions/wp_cache_get/
+		 *
+		 * Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
 		 */
-		if (
-			// Ignore cache when `WP_DEBUG` is enabled, so it doesn't interfere with the theme developers workflow.
-			( ! defined( 'WP_DEBUG' ) || ! WP_DEBUG ) &&
-			( 0 === $theme_has_support || 1 === $theme_has_support )
-		) {
+		if ( ! WP_DEBUG && is_int( $theme_has_support ) ) {
 			return (bool) $theme_has_support;
 		}
 
-		// Has the own theme a theme.json?
-		$theme_has_support = is_readable( get_stylesheet_directory() . '/theme.json' ) ? 1 : 0;
-
-		// Look up the parent if the child does not have a theme.json.
-		if ( 0 === $theme_has_support ) {
-			$theme_has_support = is_readable( get_template_directory() . '/theme.json' ) ? 1 : 0;
+		$wp_theme = wp_get_theme( $stylesheet );
```

Comment:
See conversation about performance of `wp_get_theme` at https://github.com/WordPress/gutenberg/pull/45831/files#r1029225791

Reply:
This allows us to get if another theme supports theme json. It is needed for [here](https://github.com/WordPress/wordpress-develop/blob/239a6c33ec5fef9416655c36225b064b987c31cc/src/wp-includes/rest-api/endpoints/class-wp-rest-themes-controller.php#L372) and [here](https://github.com/WordPress/wordpress-develop/blob/239a6c33ec5fef9416655c36225b064b987c31cc/src/wp-includes/class-wp-theme-json-resolver.php#L423-L425).

Reply:
@felixarntz YES! In this case, we call it once and after that, we cache. This is called once per page load. Making it fine. Calling it 5k times per page is the problem.